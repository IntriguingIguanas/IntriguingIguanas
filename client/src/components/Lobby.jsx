import React from 'react'
import { hashHistory } from 'react-router'
import { connect } from 'react-redux'
import { Link } from 'react-router'
import BoardPage from './BoardPage.jsx'
<<<<<<< HEAD
import { createBoard, fetchBoards } from '../actions/Board'
import SideBar from './SideBar.jsx'
=======
import { createBoard, fetchBoards, deleteBoard } from '../actions/Board'
>>>>>>> Add delete board functions

export class Lobby extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      boardNameInput: ''
    }
    this.handleBoardNameChange = this.handleBoardNameChange.bind(this)
    this.createBoard = this.createBoard.bind(this)
    this.deleteBoard = this.deleteBoard.bind(this)
  }

  componentWillMount() {
    console.log('> LOBBY', this.props)
    console.log('> USER_ID', this.props.LogIn.user_id)
    this.props.fetchBoards(this.props.LogIn.user_id)
  }


  handleBoardNameChange(e) {
    this.setState({
      boardNameInput: e.target.value
    })
  }

  createBoard(e) {
    e.preventDefault()
    this.props.createBoard(this.state.boardNameInput, this.props.LogIn.user_id)
  }

  deleteBoard(board_id) {
    this.props.deleteBoard(board_id, this.props.LogIn.user_id)
  }

  render() {
    console.log('> BOARDS', this.props.boards)
    return (
      <div>
        <input
          value={ this.state.boardNameInput }
          onChange={ this.handleBoardNameChange }
        />
        <button onClick={ this.createBoard }>CREATE BOARD</button>
        { this.props.boards.map((board) => (
            <div>
              <Link to={`/lobby/${board.boardname}/${board.id}`}>
              { board.boardname }
              </Link>
              <button onClick={ () => { this.deleteBoard(board.id) } }>Delete</button>
            </div>
        )) }
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    ...state,
    boards: state.board.boards,
    user_id: state.LogIn.user_id

  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    createBoard: (boardName, user_id) => { dispatch(createBoard(boardName, user_id)) },
    deleteBoard: (board_id, user_id) => { dispatch(deleteBoard(board_id, user_id)) },
    fetchBoards: (user_id) => { dispatch(fetchBoards(user_id)) }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Lobby)