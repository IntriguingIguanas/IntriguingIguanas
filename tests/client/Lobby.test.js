import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import { Lobby } from 'Lobby'

describe('Lobby', () => {
  var boardList = [{boardName: "BOARD 1"}, {boardName: "BOARD 2"}]
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Lobby boardlist={ boardList }/>, div);
  })
  it('should render create board input', () => {
    const lobby = renderer.create(<Lobby boardlist={ boardList } />).toJSON();
    expect(lobby).toMatchSnapshot();
  })
})