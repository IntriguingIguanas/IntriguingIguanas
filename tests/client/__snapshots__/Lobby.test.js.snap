// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lobby should render create board input 1`] = `
<div>
  <input
    onChange={[Function]}
    value=""
  />
  <button
    onClick={[Function]}
  >
    CREATE BOARD
  </button>
  <div
    onClick={[Function]}
  >
     
    BOARD 1
  </div>
  <div
    onClick={[Function]}
  >
     
    BOARD 2
  </div>
</div>
`;
